### ISSUE TITLE
Replace fallback DNS with recursive DNS resolver

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#5853]
We've discussed adding a recursive DNS resolver to tailscaled for various things. Originally @danderson proposed it as a means to do DNS resolution when we're in a sandbox and don't have permission to read the host operating system's DNS settings and we don't want to hard-code any public well-known recursive resolver.

One thing we could use a recursive DNS resolver for is replacing our fallbackdns (bootstrap DNS) so we don't need use our existing DERP servers.

/cc @DentonGentry @maisem @andrew-d @crawshaw

**Comment by danderson** [comment]
The only scenario I can think of where that won't work out, is on networks that block udp/53 traffic to everything except the local resolver. Right now in those networks, with a misconfigured client, we're still able to get out to the internet over HTTPS and get a good resolution to fix up the local config. A recursive resolver would fail and leave the machine stranded.

OTOH, it would save us from the headaches of having hardcoded bootstrap maps in old binaries, and having to maintain moribund servers for ever to keep serving those clients.

**Comment by bradfitz** [comment]
Another thing to consider: we have IP space we own these days and could start anycasting it for bootstrap DNS. Then at least we're only baking in one/two IPs. It doesn't have to even be particularly good anycast if it's only for bootstrap DNS.

**Comment by danderson** [comment]
Yeah, anycasting to our derps (or something else) would help future-proof things. It means we're committed to owning our AS and IP space "for ever" but that's significantly easier to do.

**Comment by DentonGentry** [comment]
Related: https://github.com/tailscale/tailscale/issues/3507

**Comment by DentonGentry** [comment]
(we used to have an entirely separate concept of Fallback Resolvers and it keeps weirding me out)
